db/schema.rb

```diff
@@ -786,22 +808,26 @@
     t.date     "sub_2"
     t.string   "dry_book"
     t.integer  "login_count"
-    t.datetime "created_at",                          null: false
-    t.datetime "updated_at",                          null: false
-    t.string   "email",                  default: "", null: false
-    t.string   "encrypted_password",     default: "", null: false
+    t.datetime "created_at",                           null: false
+    t.datetime "updated_at",                           null: false
+    t.string   "email",                   default: "", null: false
+    t.string   "encrypted_password",      default: "", null: false
     t.string   "reset_password_token"
     t.datetime "reset_password_sent_at"
     t.datetime "remember_created_at"
-    t.integer  "sign_in_count",          default: 0,  null: false
+    t.integer  "sign_in_count",           default: 0,  null: false
     t.datetime "current_sign_in_at"
     t.datetime "last_sign_in_at"
     t.inet     "current_sign_in_ip"
     t.inet     "last_sign_in_ip"
+    t.integer  "tsheets_user_id"
```

Comment:
Tsheets_ columns should probably be here, right?

Reply:
I hate this, do you have a good way of dealing with it? I've seen people use git hooks to rollback migrations when changing branches then run migrations in the new branch. It seems a bit heavy handed, but not sure if there's a better way?

Reply:
Those migration scripts are a bit strange, Sam. I usually resort to building database from schema.rb (or structure.sql). You should be able to rollback last migration with `rake db:rollback STEP=1`. Or just drop column manually.

Reply:
Well dropping the column manually is not a good idea because then when I check out the tsheets branch, the column is not there, but it thinks the migrations have run. Building the db from the schema wouldn't work at that point either because the schema is wrong. You could drop, create and migrate, but we'd need decent seed data to be able to run that often. It really seems we're left with rolling back any migrations whenever you leave a branch which seems a lot to remember.

Honestly I've never found a good solution for this. Those migration scripts are indeed a bit strange - they would solve the issue, but something about it doesn't feel quite right.

Reply:
>Well dropping the column manually is not a good idea because then when I check out the tsheets branch, the column is not there, but it thinks the migrations have run. 

@cheeseweasel if you drop manually column from database, then you also need to remove your migration id from `schema_migrations` table.

Reply:
> You could drop, create and migrate, but we'd need decent seed data to be able to run that often.

Yeah, i've added separate seed_migration functionality. But so far, i'm not sure that seed data is correct as you said. I personally feel that there are too much lookup tables, that make this seed data a special kind of nightmare. It would be nice to slowly remove database lookup tables and just store lookups as hash -- it will be faster, one less database call (or a join) and it will be easier for everyone to grasp as well :-\